### DISCUSSION TITLE
Why arrays not working :(

### DISCUSSION THREAD

**Discussion by mhoeschele-is-taken** [eliasm307/storybook-addon-deep-controls#21] [Q&A] [answered]
I wanted to ask whats the reason behind arrays not working was it just bc u didnt want to or bc there was an unfixable problem?
Is there maybe another solution like yours with arrays XD
i could need sum help :/

**Comment by eliasm307** [comment]
The reason for arrays being left as is is because if we flattened the array, eg  considering the array as an object with number properties, then it means there is no way to modify the array eg add a new item to the array, so users would lose some functionality

With this addon the aim was to make things easier when working with non-primitive controls and not reduce functionality as much as possible, and being able to modify arrays seemed like an important thing to keep as there would be no workaround if it is important.

If we flattened arrays, then it might be a case of users need to disable the addon for stories where modifying arrays is important and that sounded annoying to me

**Reply by eliasm307** [reply]
@mhoeschele-is-taken what is your use case, I could provide a way to explicitly allow certain arrays to be flattened if that is important?

**Reply by mhoeschele-is-taken** [reply]
right now the problem im facing is that i have a footer component with json-data that roughly looks something like
`Default.args = {
    a: {
        aa: [ funny array stuff ],
        ab: [ more funny array stuff ],
        ac: [ (a)rey skywalker ],
    },
    b: [ array explosion ]
}`

which means i have multiple arrays inside a key but also an array at the top level. its not that common that i want to add more arrays but if i do i got alternative methods so the "adding new indexes" aspect is not that important

**Reply by mhoeschele-is-taken** [reply]
i found out what spreading is
i can just go
const obj = {...array};
and how to reverse it
const array = Object.value(obj);

what i dont get is the part in the middle where i want to make controls automatically and inject the changes into my args/managing where the code should be used

im quite the beginner so pls dont be mad at me

**Reply by mhoeschele-is-taken** [reply]
OK little Update on my progress:
I managed to implement a solution that turns every array into an object before giving the data to deep control
and changes it back before sending it to my component. It has a 0.5 sec delay after every input but it works "i see that as an absolute win" xd

and now i have 700 controls in this one story....
im trying to figure out how to 'table : { category: "" }}`
them now with regex

edit:
got it to work by writing a function that takes all the names of the controls etc.
then it scans them with regex rules i set
```const rules = [
	{regex: 'footerColsContent\\.colPos0\\.*', category: 'Col 0', subcategory: 'Sub Col 0'},
	{regex: 'footerColsContent\\.colPos1\\.*', category: 'Col 1', subcategory: 'Sub Col 1'},
];
const categorizedArgs = spreader.categorizeArgs(args, rules);
```
the output i get is
```
'footerColsContent.colPos1."""a new object for every name""": {
  table: {
    category: 'Col 1',
    subcategory: 'Sub Col 1',
  },
},
'footerColsContent.colPos0."""a new object for every name""": {
  table: {
    category: 'Col 0',
    subcategory: 'Sub Col 0',
  },
},
```

and then put it into the argTypes by going:
```
argTypes: {...categorizedArgs},
```  

NOOOW the problem is that it overwrites everything deep control did :cry:

**Reply by eliasm307** [reply]
Hey, I don't fully understand your example, can you provide a simple example of a working component and story that demonstrates the issue then I can look into it?

**Reply by mhoeschele-is-taken** [reply, marked as answer]
Hey sorry for the late answer.
Im sadly not allowed to talk about this any further.
Just know i managed to find a solution that works for me.
Sorry for takin up your time.
